WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -216,4 +216,13 @@ def export_public
   def countries
     render json: Country.all
   end
+
+  def competition_series
+    competition_series = CompetitionSeries.find_by_wcif_id(params[:id])
+    if competition_series&.all_competitions_public?
```

Comment:
Cool that you thought about this. I wonder whether it might be a better approach to just filter out the competitions that are not public, and then perhaps throw some error (like pretend the series was not found) if the filtered list is empty

Reply:
Addressed [below](https://github.com/thewca/worldcubeassociation.org/pull/8125#issuecomment-1657184645) and in following commits. Thanks for the suggestion.